beacon-chain/powchain/service.go

```diff
@@ -725,3 +727,20 @@ func (s *Service) logTillChainStart() {
 		"Time till minimum genesis": time.Duration(secondsLeft) * time.Second,
 	}).Infof("Currently waiting for chainstart")
 }
+
+// cacheHeadersForEth1DataVote makes sure that voting for eth1data after startup utilizes cached headers
+// instead of making multiple RPC requests to the ETH1 endpoint.
+func (s *Service) cacheHeadersForEth1DataVote(lastKnownHeader *gethTypes.Header) {
+	blocksPerVotingPeriod := params.BeaconConfig().EpochsPerEth1VotingPeriod * params.BeaconConfig().SlotsPerEpoch *
+		params.BeaconConfig().SecondsPerSlot / params.BeaconConfig().SecondsPerETH1Block
+
+	end := lastKnownHeader.Number.Uint64() - params.BeaconConfig().Eth1FollowDistance
```

Comment:
Check for underflow, even though this cant happen after block height 1024, it can mess up some people running dev goerli chains